src/clj/rems/db/user_settings.clj

```diff
@@ -25,20 +26,14 @@
       validate-user-settings
       json/generate-string))
 
-(s/defschema PartialUserSettings
-  (map-keys s/optional-key UserSettings))
-
-(def ^:private coerce-partial-user-settings
-  (coerce/coercer! PartialUserSettings json/coercion-matcher))
+(def ^:private coerce-user-settings
```

Comment:
After getting rid of PartialUserSettings I think you can also remove the `(= (set (keys validated)) (set (keys new-settings)))` validation from `update-user-settings!`, right?

Reply:
Well reading the code now. I think the key comparison is there so that all your "settings" must be valid or the entire request is considered invalid. So if you try to set with an invalid-looking email address, the entire request will fail, no matter that it passes the schema and any other properties are ok.

Reply:
Oh right, I see. Maybe `validate-new-settings` could just return nil if any of the validations fail? And if that's logic is difficult to write neatly, you can just push the key set check inside `validate-new-settings`, right?

Reply:
I agree that it'd be more readable if the validation actaully had an easy to check failure result.